### PR TITLE
proxy: make --cipher-suites take effect for the proxy

### PR DESCRIPTION
**Pull request by rkday**
This fixes https://github.com/etcd-io/etcd/issues/10659.

It'd be great if this could be back-applied to 3.3 as well as 3.4.

### CODE REVIEW

File: etcdmain/etcd.go [hunk 1]
```diff
@@ -527,6 +527,24 @@ func startProxy(cfg *config) error {
...
 
+	if len(cfg.ec.CipherSuites) == 0 {
+		if lg != nil {
+			lg.Info("Using default cipher suites")
+		} else {
+			plog.Info("Using default cipher suites")
+		}
+	} else {
+		if lg != nil {
+			lg.Info("Using cipher suites:", zap.Strings("cipher-suites", cfg.ec.CipherSuites))
```
Comment (spzala) [inline etcdmain/etcd.go]: @rkday besides fixing a fmt error as in build, I would suggest to use lower case for all the log messages (using vs Using ..) per convention. Thanks!

**Comment by rkday** [comment]
OK, I've fixed the format error and the log text. There are a couple of test failures but I really can't see how they could be related - is there a chance they're spurious?

**Comment by jingyih** [comment]
Looks like the failing test are unrelated, don't worry about it.

**Comment by spzala** [comment]
@rkday I have mentioned it in https://github.com/etcd-io/etcd/pull/10682 which seems a dup PR but fyi to take a look. Thanks!

**Comment by spzala** [comment]
> OK, I've fixed the format error and the log text. There are a couple of test failures but I really can't see how they could be related - is there a chance they're spurious?

Thanks for addressing the comments @rkday

**Comment by rkday** [comment]
Both this and #10682 look fairly similar - this has some extra logs, and we've made different (ultimately stylistic) choices about whether UpdateCipherSuites should be exported or not. I don't really mind which gets merged, but it would be good to have one of them in soon.

**Comment by spzala** [comment]
@rkday yup, thanks and sounds good. @hexfusion is looking at it as he commented in the other PR, we can go with his decision :-) Thanks!

**Pull request closed without merging**